Backend/Tumultu/Application/Common/Interfaces/IRepository.cs

```diff
@@ -1,14 +1,22 @@
-﻿using Tumultu.Domain.Common;
+using Tumultu.Domain.Common;
 
-namespace Tumultu.Application.Common.Interfaces
+namespace Tumultu.Application.Common.Interfaces;
+
+public interface IRepository<TEntity, TId> where TEntity : BaseEntity<TId>
 {
-    public interface IRepository<TEntity, TKey> where TEntity : BaseEntity<TKey> 
-    {
-        Task<ICollection<TEntity>> GetAllAsync();
-        Task<TEntity> GetByIdAsync(TKey id);
-        Task<TEntity> AddAsync(TEntity entity);
-        Task<TEntity> UpdateAsync(TEntity entity);
-        Task<TEntity> DeleteAsync(TKey id);
-        Task<TEntity> FindAsync(TKey id);
-    }
-}
+    Task<IEnumerable<TEntity>> GetAllAsync();
```

Comment:
I am still not sure if on repository we should have query methods (GetAll, GetById) and if we should maybe put them only in readonly repository to force a convention